galaxy_ml/keras_galaxy_models.py

```diff
@@ -876,7 +876,12 @@ def fit(self, X, y, class_weight=None, **kwargs):
         -----------
         X : array-like, shape `(n_samples, feature_arrays)`
         """
-        X, y = check_X_y(X, y, accept_sparse=['csr', 'csc'], allow_nd=True)
+        multi_output = False
+        if y.ndim == 2:
+            multi_output = True
+
+        X, y = check_X_y(X, y, accept_sparse=['csr', 'csc'], allow_nd=True,
+                         multi_output=multi_output)
         check_classification_targets(y)
```

Comment:
Can we just let `multi_output=True` for this? If the type of classification problem needs to be checked, sklean provides couple of utils to call, such as `type_of_target`, 'is_multilabel' and so on.
https://github.com/scikit-learn/scikit-learn/blob/8ea176ae0ca535cdbfad7413322bbc3e54979e4d/sklearn/utils/multiclass.py#L186
https://github.com/scikit-learn/scikit-learn/blob/8ea176ae0ca535cdbfad7413322bbc3e54979e4d/sklearn/utils/multiclass.py#L113

Reply:
Also, there is another `check_X_y` under the `KerasGBatchClassifier`. Could you please change that as well?

Reply:
Will change that one too.

Reply:
> Can we just let `multi_output=True` for this? If the type of classification problem needs to be checked, sklean provides couple of utils to call, such as `type_of_target`, `is_multilabel` and so on.
> https://github.com/scikit-learn/scikit-learn/blob/8ea176ae0ca535cdbfad7413322bbc3e54979e4d/sklearn/utils/multiclass.py#L186
> https://github.com/scikit-learn/scikit-learn/blob/8ea176ae0ca535cdbfad7413322bbc3e54979e4d/sklearn/utils/multiclass.py#L113

What I did was to make sure the existing code work as before (Flag remains False), and only change the flag if output data is multidimensional (New use case).